src/collections/utils.ts

```diff
@@ -37,7 +37,7 @@ interface ParsedUri {
  */
 export const parseUri = (uri: string): ParsedUri => {
   const parsedUrl = url.parse(uri, true);
-  const baseUrl = `${parsedUrl.protocol}//${parsedUrl.host}`;
```

Comment:
This is taken care in httpClient. In the url we pass, we will have to give the `baseApiPath` as a query parameter.

```
if(options.baseApiPath){
      this.baseUrl = this.baseUrl + "/" + options.baseApiPath;
}
```
For example: http://localhost:8080/v1/testks1?baseApiPath=v1

Reply:
Could you just try giving this query parameter without this fix and see if that works?

Reply:
This can also be given as a connection option similar to username/password as `{'baseApiPath': 'v1'}`.
The reason why I set this as an option/param is that Astra/Postman url might be slightly different. For eg. a Postman mock url is as following: https://7ef39132-f969-11bf-a46a-683df0c7ce6c.mock.pstmn.io/testks1 and so here we don't have to append '/v1' to the url.

Reply:
Is there any way we can default `baseApiPath` to 'v1'? It does make sense to have the 'baseApiPath' option given needing to test against postman, but it seems a bit unnecessary to ask the end user to set that.

Reply:
@kathirsvn I made `baseApiPath` default to 'v1' in cdd562f and confirmed that seems to work fine in the sample app. Can you please take a look and see if that works for you? You should still be able to unset the `baseApiPath` by setting `baseApiPath: ''` for postman testing, because `??` only catches null and undefined, not empty string.

Reply:
Hi @vkarpov15
I agree with the concern that its bit unnecessary to ask the end to set the `baseApiPath`.

But, instead of setting it to 'v1' by default, I have added a small fix (in this commit [27fde10](https://github.com/stargate/stargate-mongoose/pull/32/commits/27fde105a92d6acc7b5931d50b29d6e7b58a8243)) to get the `baseApiPath` from the URL itself and honor the options override when present (options override may not be required in most cases, just leaving it there to override when required in some rare scenario).

This handles both empty and different variations of the `baseApiPath`. I have added some test cases as well. Could you please review and let me know if this looks good?